tests/js/client/shell/shell-transaction-intermediate-commit-cluster.js

```diff
@@ -0,0 +1,491 @@
+    }
+    internal.sleep(0.25);
+  }
+  assertEqual(leaderChecksum, followerChecksum);
+}
+
+function transactionIntermediateCommitsSingleSuite() {
+  'use strict';
+  const cn = 'UnitTestsTransaction';
+
+  return {
+
+    setUp: function () {
+      getEndpointsByType("dbserver").forEach((ep) => debugClearFailAt(ep));
+      db._drop(cn);
+    },
+
+    tearDown: function () {
+      getEndpointsByType("dbserver").forEach((ep) => debugClearFailAt(ep));
+      db._drop(cn);
+    },
+    
+    // make follower execute intermediate commits (before the leader), but let the
+    // transaction succeed
+    testSingleAqlIntermediateCommitsOnFollower: function () {
+      db._create(cn, { numberOfShards: 1, replicationFactor: 2 });
+      let shards = db._collection(cn).shards(true);
+      let shardId = Object.keys(shards)[0];
+      let leader = getEndpointById(shards[shardId][0]);
+      let follower = getEndpointById(shards[shardId][1]);
+      // disable intermediate commits on leader
+      debugSetFailAt(leader, "noIntermediateCommits");
+      // turn on intermediate commits on follower
+      debugClearFailAt(follower, "noIntermediateCommits");
+      
+      let droppedFollowersBefore = getMetric(leader, "arangodb_dropped_followers_count");
+      let intermediateCommitsBefore = getMetric(follower, "arangodb_intermediate_commits");
+      db._query('FOR i IN 1..10000 INSERT { _key: CONCAT("test", i) } IN ' + cn, {}, {intermediateCommitCount: 1000});
+
+      // follower must not have been dropped
+      let droppedFollowersAfter = getMetric(leader, "arangodb_dropped_followers_count");
+      assertEqual(droppedFollowersBefore, droppedFollowersAfter);
+      
+      let intermediateCommitsAfter = getMetric(follower, "arangodb_intermediate_commits");
+      assertEqual(intermediateCommitsBefore + 10, intermediateCommitsAfter);
+      
+      assertInSync(leader, follower, shardId);
+    },
+    
+    // make follower execute intermediate commits (before the leader), and let the
+    // transaction fail
+    testSingleAqlIntermediateCommitsOnFollowerWithRollback: function () {
+      db._create(cn, { numberOfShards: 1, replicationFactor: 2 });
+      let shards = db._collection(cn).shards(true);
+      let shardId = Object.keys(shards)[0];
+      let leader = getEndpointById(shards[shardId][0]);
+      let follower = getEndpointById(shards[shardId][1]);
+      // disable intermediate commits on leader
+      debugSetFailAt(leader, "noIntermediateCommits");
+      // turn on intermediate commits on follower
+      debugClearFailAt(follower, "noIntermediateCommits");
+      
+      let droppedFollowersBefore = getMetric(leader, "arangodb_dropped_followers_count");
+      let intermediateCommitsBefore = getMetric(follower, "arangodb_intermediate_commits");
+      try {
+        db._query('FOR i IN 1..10000 INSERT { _key: CONCAT("test", i), value: ASSERT(i < 10000, "peng!") } IN ' + cn, {}, {intermediateCommitCount: 1000});
+        fail();
```

Comment:
This pattern does not work!
`fail()` will throw an error if triggered. You need this error to be passed through.
But it will be watched in the catch all. This way you miss if the query fails or not.